### PR TITLE
[hueemulation] Fix ipv4 or ipv6 only environment

### PR DESCRIPTION
**Pull request by ngrodum**
This PR has two slightly different fixes for the same problem, I have split them into separate commits.

channel.open would throw if ipv6 was not enabled. This would cause ipv4 to fail as well.

     * @throws  UnsupportedOperationException
     *          If the specified protocol family is not supported. For example,
     *          suppose the parameter is specified as {@link
     *          java.net.StandardProtocolFamily#INET6 StandardProtocolFamily.INET6}
     *          but IPv6 is not enabled on the platform.

I also took the liberty of changing the boolean flags to just use the channels directly. I don't think a lot of readability is lost in the process and seems quite redundant. Also, I got some codestylewarnings when the boolean flags were used - they weren't treated as strong enough nullpointer-exception guards.

### CODE REVIEW

File: bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/upnp/UpnpServer.java [hunk 1]
```diff
@@ -409,48 +409,47 @@ private void sendUPNPNotify(DatagramSocket sendSocket, InetAddress address, int
...
     public void accept(HueEmulationConfigWithRuntime threadContext) {
         logger.info("Hue Emulation UPNP server started on {}:{}", threadContext.addressString, threadContext.port);
-        boolean hasIPv4 = false;
-        boolean hasIPv6 = false;
 
-        try (
-                DatagramChannel channelV4 = DatagramChannel.open(StandardProtocolFamily.INET);
-                DatagramChannel channelV6 = DatagramChannel.open(StandardProtocolFamily.INET6);
-                Selector selector = Selector.open()) {
+        try (Selector selector = Selector.open()) {
```
Comment (J-N-K) [inline bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/upnp/UpnpServer.java]: I understand why you removed the channels here, but maybe you could add a `finally` clause and free the resources there.

**Comment by ngrodum** [comment]
For reference, here is a stack trace I would get if I restarted the hueemulation binding:

Exception in thread "HueEmulation UPNP Server" java.lang.UnsupportedOperationException: IPv6 not available
at sun.nio.ch.DatagramChannelImpl.<init>(DatagramChannelImpl.java:138)
at sun.nio.ch.SelectorProviderImpl.openDatagramChannel(SelectorProviderImpl.java:46)
at java.nio.channels.DatagramChannel.open(DatagramChannel.java:182)
at org.openhab.io.hueemulation.internal.upnp.UpnpServer.dataGramChannelOrNull(UpnpServer.java:500)
at org.openhab.io.hueemulation.internal.upnp.UpnpServer.accept(UpnpServer.java:417) at org.openhab.io.hueemulation.internal.upnp.UpnpServer.accept(UpnpServer.java:1) at org.openhab.io.hueemulation.internal.upnp.HueEmulationConfigWithRuntime.run(HueEmulationConfigWithRuntime.java:111)

File: bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/upnp/UpnpServer.java (hunk 1 above)
Comment (ngrodum) [reply bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/upnp/UpnpServer.java]: Yeah it needs some fixing - there seems to be some multithreading going on too by looking at the failed Travis test.

Comment (ngrodum) [reply bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/upnp/UpnpServer.java]: @J-N-K  I rewrote the change, hope you're ok with me edit/force pushing in my branch, it's a small commit..

File: bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/upnp/UpnpServer.java [hunk 2]
```diff
@@ -61,6 +61,7 @@
 import org.osgi.service.http.NamespaceException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import sun.nio.ch.Net;
```
Comment (davidgraeff) [inline bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/upnp/UpnpServer.java]: Hm. "sun" headers are generally deprecated I think. Is there an alternative in the JDK?

**Comment by davidgraeff** [comment]
The upnp server exception is a race condition that only occurs on specific load conditions. Would be sweet to have that fixed, but it is unrelated to this change.

File: bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/upnp/UpnpServer.java (hunk 2 above)
Comment (ngrodum) [reply bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/upnp/UpnpServer.java]: No idea how it snuck in there. It was unused

File: bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/upnp/UpnpServer.java (hunk 1 above)
Comment (davidgraeff) [reply bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/upnp/UpnpServer.java]: @J-N-K Are you fine with the current use of try-resources? Would merge if no objections

**Pull request merged**